osquery/main/tests.cpp

```diff
@@ -1,20 +1,124 @@
 /*
- *  Copyright (c) 2014-present, Facebook, Inc.
- *  All rights reserved.
- *
- *  This source code is licensed under the BSD-style license found in the
- *  LICENSE file in the root directory of this source tree. An additional grant
- *  of patent rights can be found in the PATENTS file in the same directory.
- *
- */
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#ifdef WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <Shlwapi.h>
+#endif
+
+#include <memory>
 
 #include <gtest/gtest.h>
 
 #include <osquery/logger.h>
 
+#include "osquery/core/process.h"
 #include "osquery/core/test_util.h"
 
+#define OSQUERY_TESTS_MODULENAME  "osquery_tests.exe"
+
+char *self_exec_path = nullptr;
+
+const char *expected_worker_args[] = {
+  "worker-test"
+};
+const char *expected_extension_args[] = {
+  "osquery extension: extension-test",
+  "--socket",
+  "socket-name",
+  "--timeout",
+  "100",
+  "--interval",
+  "5",
+  "--verbose"
+};
+
+bool compareArguments(char *result[],
+  unsigned int result_nelms,
+  const char *expected[],
+  unsigned int expected_nelms) {
+  if (result_nelms != expected_nelms) {
+    return false;
+  }
+
+  for (unsigned int i = 0; i < expected_nelms; i++) {
+    if (strlen(result[i]) != strlen(expected[i])) {
+      return false;
+    }
+
+    if (strncmp(result[i], expected[i], strlen(expected[i])) != 0) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+int workerMain(int argc, char *argv[]) {
+  if (!compareArguments(argv,
+    argc,
+    expected_worker_args,
+    sizeof(expected_worker_args) / sizeof(const char *))) {
+    return -1;
+  }
+
+  std::shared_ptr<osquery::PlatformProcess> process = osquery::getLauncherProcess();
+  if (!process) {
```

Comment:
Not sure what this means? Prefer to be clean and clear, compare to what you expect and do not expect. If you want to return `-2` if you received a `nullptr` please compare against a `nullptr` this helps dramatically with static analysis tools, as well as provides easier flow/understanding for future developers.